pkgs/data/misc/hackage/default.nix

```diff
@@ -1,6 +1,6 @@
-{ fetchurl }:
-
-fetchurl {
-  url = "https://github.com/commercialhaskell/all-cabal-hashes/archive/6515ef12bbcf8fbac87e12b4cb30b7eefa9ce9ce.tar.gz";
-  sha256 = "0plf0kk0wj1lbmks09afyqrl70z0miwxzfk3zh7y2qiw3g5l1v0x";
-}
+# Hackage database snapshot, used by maintainers/scripts/regenerate-hackage-packages.sh
+{ fetchFromGitHub }:
+let
+  pin = builtins.fromJSON (builtins.readFile ./pin.json);
+in
+  fetchFromGitHub (pin // { passthru.commit = pin.rev; })
```

Comment:
@hyperfekt I was under the impression that this file was getting updated automatically every so often.

If you change it's format here, I can imagine that it will no longer be able to be updated automatically.

Reply:
There was an update from time to time but it was not in sync with the version used to generate hackage-packages.nix. I opened NixOS/cabal2nix#456 and NixOS/cabal2nix#457 to make sure that is the case in the future.

Reply:
In fact it is essential that either one of those PRs is merged if this one is because otherwise the script will not be able to find the versions used for generating the current hackage-packages.nix,
as I really wanted to avoid parsing the git commit messages in it.